### PR TITLE
GSE2 bulletin: Fix reading station magnitude and amplitude

### PR DESCRIPTION
**Pull request by thefroid**
### What does this PR do?

The plugin _read_gse2 was not able to read Mag2 for each station if a Mag1 was not defined.
I corrected it to read Mag2 for each station even if Mag1 is not defined.
Moreover, an amplitude object is not created if information about magnitude is not present,
more explicitly i don't fetch amplitude, period... if we don't know what type of amplitude is computed.

### Why was it initiated?  Any relevant Issues?

*Please fill in*

### PR Checklist
- [x] Correct base branch selected? `master` for new features, `maintenance_...` for bug fixes
- [ ] This PR is not directly related to an existing issue (which has no PR yet).
- [ ] If the PR is making changes to documentation, docs pages can be built automatically.
      Just remove the space in the following string after the + sign: "+ DOCS"
- [ ] If any network modules should be tested for the PR, add them as a comma separated list
      (e.g. `clients.fdsn,clients.arclink`) after the colon in the following magic string: "+TESTS:"
      (you can also add "ALL" to just simply run all tests across all modules)
- [ ] All tests still pass.
- [x] Any new features or fixed regressions are be covered via new tests.
- [ ] Any new or changed features have are fully documented.
- [x] Significant changes have been added to `CHANGELOG.txt` .
- [ ] First time contributors have added your name to `CONTRIBUTORS.txt` .

### CODE REVIEW

File: obspy/io/gse2/bulletin.py [hunk 1]
```diff
@@ -896,18 +896,24 @@ def _parse_arrivals(self, event, origin, origin_res_id):
...
+                    try:
+                        sta_mag.mag = magnitude_values[i]
+                    except ValueError:
+                        continue
+                    flag_sta_mag = True
                     public_id = "magnitude/station/%s/%s" % (line_id, i)
                     sta_mag.resource_id = self._get_res_id(public_id)
                     event.station_magnitudes.append(sta_mag)
+                if flag_sta_mag:
+                    event.amplitudes.append(amplitude)
```
Comment (megies) [inline obspy/io/gse2/bulletin.py]: Does it hurt to add the amplitude even if no station magnitude is using it?

File: obspy/io/gse2/tests/test_bulletin.py [hunk 2]
```diff
@@ -2,7 +2,6 @@
 # -*- coding: utf-8 -*-
 """
 The gse2.bulletin test suite.
-
```
Comment (megies) [inline obspy/io/gse2/tests/test_bulletin.py]: This blank should stay in

**Comment by megies** [comment]
Right now, any PR should be based against master.

**Comment by thefroid** [comment]
Ok it's not a problem, i added the amplitude even if there is no station magnitude.

**Comment by megies** [comment]
2 pep8 fails, see circle CI log at bottom

**Comment by megies** [comment]
I felt free to rebase this and force push @thefroid so we can finalize this

File: obspy/io/gse2/tests/test_bulletin.py [hunk 3]
```diff
@@ -323,6 +357,8 @@ def test_amplitude(self):
         catalog = _read_gse2(filename)
         self.assertEqual(len(catalog), 1)
         amplitudes = catalog[0].amplitudes
+        # test a new feature: don't store an object amplitude if the magnitude
+        # type is not defined
```
Comment (megies) [inline obspy/io/gse2/tests/test_bulletin.py:361]: Is there anything missing here, @thefroid? I see this comment but the test isn't changed..

**Comment by megies** [comment]
This can be merged I think, I'm only a bit confused about the above comment in an unchanged test case

**Comment by megies** [comment]
Oh and changelog one-liner is still missing

**Comment by megies** [comment]
Rebased and added changelog

**Pull request merged**